server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalExtendedStats.java

```diff
@@ -58,12 +61,21 @@ public InternalExtendedStats(StreamInput in) throws IOException {
         super(in);
         sumOfSqrs = in.readDouble();
         sigma = in.readDouble();
+        //TODO:  While merging should make sure that Version check is appropriate
```

Comment:
I *think* what you have will already work but it'd be worth adding a unit test that reduces where one of the results has a NaN `m2`.